source/core/src/main/com/deco2800/game/components/player/PlayerMeleeAttackComponent.java

```diff
@@ -0,0 +1,275 @@
+package com.deco2800.game.components.player;
+
+import com.badlogic.gdx.math.Vector2;
+import com.badlogic.gdx.physics.box2d.*;
+import com.deco2800.game.components.CombatStatsComponent;
+import com.deco2800.game.components.Component;
+import com.deco2800.game.components.PlayerCombatStatsComponent;
+import com.deco2800.game.entities.Entity;
+import com.deco2800.game.physics.BodyUserData;
+import com.deco2800.game.physics.PhysicsLayer;
+import com.deco2800.game.physics.components.PhysicsComponent;
+import com.deco2800.game.physics.components.PhysicsComponent.AlignX;
+import com.deco2800.game.physics.components.PhysicsComponent.AlignY;
+import com.deco2800.game.rendering.AnimationRenderComponent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PlayerMeleeAttackComponent extends Component {
+    private static final Logger logger = LoggerFactory.getLogger(PlayerMeleeAttackComponent.class);
+
+    private final FixtureDef fixtureDef;
+    private Fixture fixture;
+    private PlayerCombatStatsComponent playerCombatStats;
+    private PlayerMeleeAttackComponent playerAttackComponent;
+    private boolean meleeAttackClicked;
+    private boolean closeToAttack;
+    short targetLayer = PhysicsLayer.NPC;
+
+    public PlayerMeleeAttackComponent() {
+        fixtureDef = new FixtureDef();
+    }
+
+    @Override
+    public void create() {
+        if (fixtureDef.shape == null) {
+            logger.trace("{} Setting default bounding box", this);
+            fixtureDef.shape = makeBoundingBox();
+        }
+
+        Body physBody = entity.getComponent(PhysicsComponent.class).getBody();
+        fixture = physBody.createFixture(fixtureDef);
+
+        // if sensor is false, NPC will not be able to collide with player's fixture
+        setSensor(true);
+
+        // event listeners to check if enemy is within range of melee attack
+        entity.getEvents().addListener("collisionStart", this::onEnemyClose);
+
+        playerCombatStats = entity.getComponent(PlayerCombatStatsComponent.class);
+        playerAttackComponent = entity.getComponent(PlayerMeleeAttackComponent.class);
+    }
+
+    /**
+     * Called when collision with any fixture has occurred and sets closeToAttack
+     * to be True. Event system will continuously check collision between objects in world
+     * If enemy is close and player has clicked the attack melee button, damage will be dealt to
+     * the NPC
+     *
+     * @param me is the fixture of current component
+     * @param other is the other fixture which current component is in contact
+     *              with
+     */
+    private void onEnemyClose(Fixture me, Fixture other) {
+
+//        if (playerAttackComponent.getFixture() != me) {
```

Comment:
If it's not needed, remove before pulling into master.
From analysis of this file, it's dead code.

Reply:
Ye, forgot about this. It is. Thanks